mappers/bluetooth_mapper/helper/helper.go

```diff
@@ -0,0 +1,231 @@
+var Token_client Token
+var ClientOpts *MQTT.ClientOptions
+var Client MQTT.Client
+
+//Token interface to validate the MQTT connection.
+type Token interface {
+	Wait() bool
+	WaitTimeout(time.Duration) bool
+	Error() error
+}
+
+//DeviceStateUpdate is the structure used in updating the device state
+type DeviceStateUpdate struct {
+	State string `json:"state,omitempty"`
+}
+
+//BaseMessage the base struct of event message
+type BaseMessage struct {
+	EventID   string `json:"event_id"`
+	Timestamp int64  `json:"timestamp"`
+}
+
+//TwinValue the struct of twin value
+type TwinValue struct {
+	Value    *string        `json:"value, omitempty"`
+	Metadata *ValueMetadata `json:"metadata,omitempty"`
+}
+
+//ValueMetadata the meta of value
+type ValueMetadata struct {
+	Timestamp int64 `json:"timestamp, omitempty"`
+}
+
+//TypeMetadata the meta of value type
+type TypeMetadata struct {
+	Type string `json:"type,omitempty"`
+}
+
+//TwinVersion twin version
+type TwinVersion struct {
+	CloudVersion int64 `json:"cloud"`
+	EdgeVersion  int64 `json:"edge"`
+}
+
+//MsgTwin the struct of device twin
+type MsgTwin struct {
+	Expected        *TwinValue    `json:"expected,omitempty"`
+	Actual          *TwinValue    `json:"actual,omitempty"`
+	Optional        *bool         `json:"optional,omitempty"`
+	Metadata        *TypeMetadata `json:"metadata,omitempty"`
+	ExpectedVersion *TwinVersion  `json:"expected_version,omitempty"`
+	ActualVersion   *TwinVersion  `json:"actual_version,omitempty"`
+}
+
+//DeviceTwinUpdate the struct of device twin update
+type DeviceTwinUpdate struct {
+	BaseMessage
+	Twin map[string]*MsgTwin `json:"twin"`
+}
+
+//DeviceTwinResult device get result
+type DeviceTwinResult struct {
+	BaseMessage
+	Twin map[string]*MsgTwin `json:"twin"`
+}
+
+// HubclientInit create mqtt client config
+func HubClientInit(server, clientID, username, password string) *MQTT.ClientOptions {
+	opts := MQTT.NewClientOptions().AddBroker(server).SetClientID(clientID).SetCleanSession(true)
+	if username != "" {
+		opts.SetUsername(username)
+		if password != "" {
+			opts.SetPassword(password)
+		}
+	}
+	tlsConfig := &tls.Config{InsecureSkipVerify: true, ClientAuth: tls.NoClientCert}
+	opts.SetTLSConfig(tlsConfig)
+	return opts
+}
+
+//MqttConnect function felicitates the MQTT connection
+func MqttConnect(mqttMode int, mqttInternalServer, mqttServer string) {
+	// Initiate the MQTT connection
+	if mqttMode == 0 {
+		ClientOpts = HubClientInit(mqttInternalServer, "eventbus", "", "")
+	} else if mqttMode == 1 {
+		ClientOpts = HubClientInit(mqttServer, "eventbus", "", "")
+	}
+	Client = MQTT.NewClient(ClientOpts)
+	if Token_client = Client.Connect(); Token_client.Wait() && Token_client.Error() != nil {
+		glog.Errorf("client.Connect() Error is %s", Token_client.Error())
+	}
+}
+
+//ChangeDeviceState function is used to change the state of the device
+func ChangeDeviceState(state string, deviceID string) {
+	glog.Infof("Changing the state of the device to %s", state)
```

Comment:
include the device id in the message as well

Reply:
done, thanks